### ISSUE TITLE
System.MissingMethodException: Method not found: 'System.__Canon NSubstitute.Arg.Any()'.

### ISSUE THREAD

**Issue by thomasvdb** [romantitov/MockQueryable#19]
I've upgraded to the latest release to have the support for Entity Framework Core 3.0 but I receive the following error when running my unit tests:

```
System.MissingMethodException: Method not found: 'System.__Canon NSubstitute.Arg.Any()'.
```

Using NSubstitute version 4.2.1.
Running on macOS.

Part of the unit test code:
```
var myQueryableMock = myList.AsQueryable().BuildMock();
myRepositorySubstitute.Get(Arg.Any<Expression<Func<MyEntity, bool>>>()).Returns(myQueryableMock);
```

**Comment by romantitov** [comment]
I've updated NSubstitute version 4.2.1. for MockQueryable and all tests are green. 
What happened in `myRepositorySubstitute.Get`? Why it required an expression?  Could you upload somewhere your test?

**Comment by thomasvdb** [comment]
The implementation of Repository contains the following:
```
public IQueryable<T> Get(Expression<Func<T, bool>> predicate)
{
    return DbSet().Where(predicate).AsQueryable();
}
```

But this shouldn't have an impact as it's mocked.
This is an example of the test:

```
        [TestMethod]
        public async Task Just_an_example_test()
        {
            const string aRandomId = "randomid";

            var accounts = new[]
            {
                new AnAccount
                {
                    Id = Guid.NewGuid(),
                    RandomId = aRandomId
                },
                new AnAccount
                {
                    Id = Guid.NewGuid(),
                    RandomId = aRandomId
                }
            };

            var accountsQueryableMock = accounts.AsQueryable().BuildMock();

            myRepositorySubstitute.Get(Arg.Any<Expression<Func<AnAccount, bool>>>())
                .Returns(accountsQueryableMock);
      
            await subject.DoSomeAction(aRandomId, accounts[1].Id);

            myRepositorySubstitute.Received().Remove(Arg.Is<AnAccount>(x =>
                x.Id == accounts[1].Id));
        }
```

The implementation of the subject:
```
        public async Task DoSomeAction(string aRandomId, Guid accountId)
        {
            var anAccount = await myRepository.Get(x => string.Equals(aRandomId,
                    x.ARandomId)).FirstOrDefaultAsync();
            myRepository.Remove(account);
        }
```

The test has been altered and naming has been changed so it doesn't make any sense :)
But the main idea is still kept.

**Comment by romantitov** [comment]
Looks like this is issue with NSubstitute` 4.2.1`  for version `3.1.0` it works. I'm not sure, if I can fix it. I would reccomend you to set up  NSubstitute `3.1.0`. Let me know, if for some objective reasons you cannot use `3.1.0`  - I can create alpha version of `MockQueryable.NSubstitute` which required NSubstitute >=` 4.2.1` .

**Comment by romantitov** [comment]
Alpha version of `MockQueryable.NSubstitute` with  NSubstitute 4.2.1 available https://www.nuget.org/packages/MockQueryable.NSubstitute/3.0.1-alpha

@thomasvdb  please let me know,  if the issue still reproducible with the version?

**Comment by romantitov** [comment]
fixed in version 3.0.1
